_episodes/l3_03_test_fixture.md

```diff
@@ -0,0 +1,491 @@
+
+In GoogleTest, a test fixture is implemented using a test fixture class. This class serves as a container for shared setup and cleanup logic, as well as any shared objects or resources required by the test cases within the fixture.
+### 1.1 When is a Test Fixture Needed?
+A test fixture is typically used in the following scenarios:
+
+1. **Shared Setup and Teardown**: When multiple test cases require a common set of setup or teardown (which means cleaning in general and is the opposite of setup) steps, a test fixture is beneficial. Instead of duplicating the setup and teardown code in each test case, we can define it once in the test fixture and reuse it across all the tests.
+
+2. **Reducing Code Duplication**: Test fixtures help in reducing code duplication. By encapsulating the common setup and teardown logic within a fixture, we avoid duplicating the same code in multiple test cases. This improves code maintainability and reduces the chances of errors due to inconsistent or incomplete setup/teardown.
+
+3. **Isolation and Independence**: Test fixtures provide a level of isolation and independence for each test case. Each test case within a fixture runs in its own instance of the fixture class, ensuring that changes made by one test case do not affect the others. This allows for parallel execution of test cases without interference.
+
+Let us understand how to create a test fixture, setup and teardown functions with examples.
+## 2. Problem under consideration for testing
+In order to create our own test fixture, we will first explain the context or problem that we are trying to solve. We will be adding tests to check that our code works as intented by making use of test fixtures.
+
+Consider that you want to write a program to manage the details of an employee. The program should allow you to add basic detaisl of emplyee such as 
+1. Name
+2. Age 
+3. Basic Salary
+4. Number of years of employment
+5. Basic Bonus the employee has received in this year.
+
+The program should calculate the Net Bonus, Tax and Salary based on the following rules.
+1. Bonus Rule: An employee gets an additional 1000£ bonus if she or he has worked for more than 10 years.
+2. Tax Rule: Tax is calculated on a combination of basic salary and net bonus as shown below.
+    
+    * `0` if salary is less than `10k` GBP.
+    * `10%` for slar between `10K-20K` GBP.
+    * `20%` for salary between `20K-50K` GBP
+    * `50%` for slary greater than `50K` GBP 
+
+Please be advised that the purpose of above rules is to create functions which can have multiple branches and hence needs additional testing. We have tried our best to keep the example simple and realistic.
```

Comment:
I don't feel this clarification is necessary. It should be understood that any examples in the course are just that, examples to prove a concept and technique. I would remove the paragraph. 
```suggestion
```